### PR TITLE
autotrash: move to pkgs/by-name and include manpage

### PR DESCRIPTION
**Pull request by MithicSpirit**
## Description of changes

Moves autotrash to pkgs/by-name, as well as building and installing its manpage. Also adds myself as the maintainer of this package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by momeemt** [comment]
Result of `nixpkgs-review pr 343855` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.autotrash</li>
    <li>python311Packages.autotrash.dist</li>
    <li>python312Packages.autotrash</li>
    <li>python312Packages.autotrash.dist</li>
  </ul>
</details>

**Review by momeemt — APPROVED** [review]
Thank you for your contribution. LGTM

**Comment by Sigmanificient** [comment]
Oh hello Mithic! :D

**Comment by MithicSpirit** [comment]
> We should probably move it into `pkgs/by-name` rather than python-modules, since it provides a app, and is separate from python ecosystem, like you said earlier
> 
> Put it under `pkgs/by-name/au/autotrash/package.nix`, and remove the entry line from `pkgs/top-level/python-packages.nix`. You'll also need to prefix some elements with `python3.pkgs` (`buildPythonPackage`, `pytestCheckHook` and python packages passed in `build-system`)

Done. I used `python3Packages` instead of `python3.pkgs` (since this doesn't directly require python3), but lmk if I should change that as well.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1035,8 +1035,6 @@ self: super: with self; {
 
   autoslot = callPackage ../development/python-modules/autoslot { };
 
-  autotrash = callPackage ../development/python-modules/autotrash { };
```
Comment (natsukium) [inline pkgs/top-level/python-packages.nix:1038]: Could you add an alias to `pkgs/top-level/python-aliases.nix`?

**Review by natsukium — CHANGES_REQUESTED** [review]

**Review by Sigmanificient — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks. Welcome to nixpkgs!🎉

**Pull request merged**